lib/SyTest/HomeserverFactory.pm

```diff
@@ -0,0 +1,39 @@
+# Copyright 2017 New Vector Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package SyTest::HomeserverFactory;
+
+# get the name of this implementation, by which it can be referenced with -I
+sub name
+{
+   my $cls = "" . shift;
```

Comment:
Stringifying the `$self` ref is probably not the best idea here. You should do that with

    my $class = ref( shift );

Reply:
it's being called as a class method, so `shift` actually gives the class name. So the forced-stringification is redundant. Removed in 3b6aecc.